websites/management/commands/markdown_cleaning/shortcode_grammar.py

```diff
@@ -0,0 +1,84 @@
+import json
+from dataclasses import dataclass
+
+from pyparsing import ParseResults, nestedExpr
+
+from websites.management.commands.markdown_cleaning.parsing_utils import WrappedParser
+
+
+@dataclass
+class ShortcodeTag:
+    """Represents a shortcode tag."""
+    name: str
+    args: "list[str]"
+    percent_delimiters: bool = False
+    closer: bool = False
+
+    def get_delimiters(self):
+        if self.percent_delimiters:
+            opening_delimiter = "{{%"
+            closing_delimiter = "%}}"
+        else:
+            opening_delimiter = "{{<"
+            closing_delimiter = ">}}"
+
+        if self.closer:
+            opening_delimiter += "/"
+
+        return opening_delimiter, closing_delimiter
+
+    def to_hugo(self):
+        """
+        Encases all shortcode arguments in double quotes, because Hugo allows it and that's simplest.
```

Comment:
Three options here:
1. Encase args in double quotes always
2. Encase args in double quotes only if necessary (if the arg contains spaces or quotes)
3. Do (2), but provide a mechanism to allow encasing other tags, too.

Originally I did (2), but that was problematic because our frontend code wants `{{< resource_link uuid "blah" >}}` even though "blah" does not really need to be encased in quotes.

(3) sounds overly complicated.

So that leaves (1), which is the simplest anyway. But our frontend code also does NOT want quotes around uuid args in resource and resource_link. But that is easy to change, so I made a PR to tolerate double quotes around the uuids. #1136